prime-router/src/main/kotlin/azure/ValidateFunction.kt

```diff
@@ -79,6 +71,47 @@ class ValidateFunction(
         }
     }
 
+    /**
+     * entry point for the /validateWithSchema endpoint, which validates a potential submission without writing
+     * to the database or storing/sending a file
+     */
+    @FunctionName("validateWithSchema")
+    @StorageAccount("AzureWebJobsStorage")
+    fun validateWithSchema(
+        @HttpTrigger(
+            name = "validateWithSchema",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS
+        ) request: HttpRequestMessage<String?>
+    ): HttpResponseMessage {
+        return try {
+            val schemaName = request.queryParameters.getOrDefault(SCHEMA_PARAMETER, null)
+            val format = request.queryParameters.getOrDefault(FORMAT_PARAMETER, null)
+            if (schemaName != null && format != null) {
+                actionHistory.trackActionParams(request)
+                val sender = getDummySender(schemaName, format)
+                if (sender != null) {
+                    processRequest(request, sender)
+                } else {
+                    return HttpUtilities.bad(
+                        request, "Could not find schema named '$schemaName"
+                    )
+                }
+            } else {
+                return HttpUtilities.bad(
+                    request, "Expected '$SCHEMA_PARAMETER' and '$FORMAT_PARAMETER' query parameters"
+                )
+            }
+        } catch (ex: Exception) {
+            if (ex.message != null) {
+                logger.error(ex.message!!, ex)
+            } else {
+                logger.error(ex)
+            }
+            HttpUtilities.internalErrorResponse(request)
+        }
+    }
+
     /**
      * Handles an incoming validation request after it has been authenticated via the /validate endpoint.
```

Comment:
Slightly pedantic, but is the second part of this comment still valid (_authenticated_)?

Reply:
Good catch! I can update this